test/integration/clean/test_clean.py

```diff
@@ -0,0 +1,157 @@
+import os
+import re
+
+import pytest
+import ttapp
+
+import utils
+
+tarantool_major_version, tarantool_minor_version = utils.get_tarantool_version()
+
+skip_cluster_cond = tarantool_major_version < 3
+skip_cluster_reason = "skip cluster instances test for Tarantool < 3"
+
+
+def check_clean(tt_app, running_targets, stop_targets, extra_setup, target, *args):
```

Comment:
Well... I agree that over-something is bad in general. The question is where we are? Already "over" or still "under"? And what is the alternative? How it affects the rest of context?

This function checks the correctness of `tt clean`. That expected prompt/messages/warnings are displayed. That affected only requested instances. This is _the essential part_ of `tt clean` functionality that must be checked at any execution of `tt clean`, so it looks quite reasonable to have such a function. The same applied to other-universal-functions.

Q: How we were living w/o it?
A: We had just a single test that checks just a couple of `tt clean` scenario (and frankly speaking checking was oversimplified)

This PR significantly extends the coverage of such instance-commands like start/stop/restart/kill/clean/logrotate by checking various combinations of initial states (which instances are running at the moment command is executed) and command target (target is omitted, app-target, instance-target). W/o this function (and similar for other commands) we would need another "over-" -- over-duplication which is much worse in terms of support and development as for me. 

One more note specific to clean-universal-function: in addition to other-universal-functions parameters (that specify already running instances and command target) in case of 'clean' we need separate parameter `stop_targets` because we need to differentiate instances that were running and then stopped and instances that haven't been started at all. One may find this part where initial setup is done a little tricky but the main part (as well as other-universal-functions) is quite simple and straightforward.

Reply:
To improve readability code that performs initial setup was separated from the checking itself.